core/dns.go

```diff
@@ -115,6 +116,41 @@ func (dnsResolver *DNSResolver) LookupTXT(hostname string) ([]string, time.Durat
 	return txt, rtt, err
 }
 
+// LookupHost uses a DNSSEC-enabled query to find all A/AAAA records associated with
+// the provided hostname. If the query fails due to DNSSEC, error will be
+// set to ErrorDNSSEC.
+func (dnsResolver *DNSResolver) LookupHost(hostname string) ([]net.IP, time.Duration, error) {
+	var addrs []net.IP
+	var answers []dns.RR
+
+	m := new(dns.Msg)
+	m.SetQuestion(dns.Fqdn(hostname), dns.TypeA)
+	r, aRtt, err := dnsResolver.LookupDNSSEC(m)
+	if err != nil {
+		return addrs, aRtt, err
+	}
+	answers = append(answers, r.Answer...)
+
+	m.SetQuestion(dns.Fqdn(hostname), dns.TypeAAAA)
+	r, aaaaRtt, err := dnsResolver.LookupDNSSEC(m)
+	if err != nil {
+		return addrs, aRtt + aaaaRtt, err
+	}
+	answers = append(answers, r.Answer...)
+
+	for _, answer := range answers {
+		if answer.Header().Rrtype == dns.TypeA {
+			a := answer.(*dns.A)
+			addrs = append(addrs, a.A)
+		} else if answer.Header().Rrtype == dns.TypeAAAA {
+			aaaa := answer.(*dns.AAAA)
+			addrs = append(addrs, aaaa.AAAA)
+		}
+	}
+
+	return addrs, aRtt + aaaaRtt, nil
```

Comment:
Side note: This is one of the cleanups I am planning for #395 after #413. Although what I've written so far (but not pushed) actually splits these into two return values so we can collect metrics for the individual DNS query types being sent (e.g. `A`/`AAAA`), not the `Lookup...` methods themselves.